### PR TITLE
Update kafka broker query

### PR DESCRIPTION
**Pull request by urso**
- on connect try to find the broker id (address must match advertised host).
- check broker is leader before querying offsets
- query offsets for all replicas
- remove 'isr' from event, and replace with boolean flag `insync_replica`
- replace `replicas` from event with per event `replica`-id

- update sarama to get offset per replica id

### CODE REVIEW

File: metricbeat/module/kafka/partition/partition.go [hunk 1]
```diff
@@ -34,66 +42,172 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
...
+
 // Fetch partition stats list from kafka
 func (m *MetricSet) Fetch() ([]common.MapStr, error) {
+	b, err := m.connect()
+	if err != nil {
+		return nil, err
+	}
 
-	response, err := m.broker.GetMetadata(&sarama.MetadataRequest{})
+	defer b.Close()
```
Comment (ruflin) [inline metricbeat/module/kafka/partition/partition.go:93]: Should we close and reconnect every time?

File: metricbeat/module/kafka/partition/partition.go [hunk 2]
```diff
@@ -34,66 +42,172 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
...
+
+				// Get oldest and newest available offsets
+				offOldest, offNewest, offOK, err := queryOffsetRange(b, id, topic.Name, partition.ID)
+
+				var offsets common.MapStr
+				if offOK {
+					offsets = common.MapStr{
+						"newest": offNewest,
+						"oldest": offOldest,
+						"error":  nil,
```
Comment (ruflin) [inline metricbeat/module/kafka/partition/partition.go:128]: should we even add error if it is nil?

File: vendor/github.com/Shopify/sarama/offset_request.go [hunk 3]
```diff
@@ -22,11 +22,20 @@ func (b *offsetRequestBlock) decode(pd packetDecoder) (err error) {
 }
 
 type OffsetRequest struct {
-	blocks map[string]map[int32]*offsetRequestBlock
+	replicaID *int32
```
Comment (ruflin) [inline vendor/github.com/Shopify/sarama/offset_request.go:25]: We need to make sure that we port these changes back to your fork of sarama

File: metricbeat/module/kafka/partition/partition.go [hunk 4]
```diff
@@ -34,66 +42,172 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
...
+func hasID(id int32, lst []int32) bool {
+	for _, other := range lst {
+		if id == other {
+			return true
+		}
+	}
+	return false
+}
+
+func queryOffsetRange(
```
Comment (ruflin) [inline metricbeat/module/kafka/partition/partition.go:171]: a doc comment would be nice here on what the different offset are that are returned.

**Comment by ruflin** [comment]
@urso Change LGTM. Tests are failing because updates to the fields are missing. Can you update this?

File: metricbeat/module/kafka/partition/partition.go (hunk 1 above)
Comment (urso) [reply metricbeat/module/kafka/partition/partition.go:93]: I think so. We query the broker every 10 seconds. No need to keep mostly idle connections.

File: metricbeat/module/kafka/partition/partition.go (hunk 2 above)
Comment (urso) [reply metricbeat/module/kafka/partition/partition.go:128]: Using nil, to keep mostly the same event format then before. We can remove the error field though.

File: metricbeat/module/kafka/partition/partition.go (hunk 4 above)
Comment (urso) [reply metricbeat/module/kafka/partition/partition.go:171]: Hehe, this one needs some more testing (like, see if it really returns correct indidces).

File: vendor/github.com/Shopify/sarama/offset_request.go (hunk 3 above)
Comment (urso) [reply vendor/github.com/Shopify/sarama/offset_request.go:25]: it's already in my fork. Isn't the glide.yaml file updated in this PR?

File: metricbeat/module/kafka/partition/partition.go (hunk 2 above)
Comment (ruflin) [reply metricbeat/module/kafka/partition/partition.go:128]: I would remove it if empty as we do the same for other fields which do not have content.

File: vendor/github.com/Shopify/sarama/offset_request.go (hunk 3 above)
Comment (ruflin) [reply vendor/github.com/Shopify/sarama/offset_request.go:25]: got it. it is updated.

**Comment by ruflin** [comment]
@urso Merging this one, so I can continue on the other PR.

**Pull request merged**